keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1108 @@
+to aggregating container requests.
+
+
+#### PodOverhead
+
+Since the PodOverhead feature works at pod-level cgroup, pod-level resource specification and PodOverhead will work together seamlessly. When the kube-scheduler is deciding 
+which node should run a new Pod, the scheduler considers that Pod's overhead as well as the sum of container requests for that Pod. Scheduler should add pod-level request,
+podoverhead to find a suitable node, when pod level request is specfied.
+
+Likewise the kubelet will need to add PodOverhead to the limits specified in the pod-level spec (instead of sum of container level requests) when setting the upper limit for
+the pod-level cgroup (i.e. cpu.max and memory.max).
+
+
+#### [Scoped for Beta] HugeTLB cgroup
+
+TBD. Do not review for the alpha stage.
+
+
+#### [Scoped for Beta] Topology Manager
+
+TBD. Do not review for the alpha stage.
+
+
+* (Tentative) Only pod level scope for topology alignment will be supported if pod level requests and limits are specified without container-level requests and limits.
+
+
+#### [Scoped for Beta] Memory Manager
+
+TBD. Do not review for the alpha stage.
+
+
+#### [Scoped for Beta] CPU Manager
+
+TBD. Do not review for the alpha stage.
+
+
+#### [Scoped for Beta] In Place Pod Resize
+
+TBD. Do not review for the alpha stage.
+
+
+#### [Scoped for Beta] VPA
+
+TBD. Do not review for the alpha stage.
+
+
+#### [Scoped for Beta] Cluster Autoscaler
+
+TBD. Do not review for the alpha stage.
+
+t
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to existing tests to make this code solid enough prior to committing the changes
+necessary to implement this enhancement.
+
+#### Unit tests
+
+This feature will touch multiple components. For alpha, unit tests coverage for following packages needs to be added:
+
+* Scheduler logic will be updated to consider pod-level requests. Hence pkg/scheduler will require additional coverage.
+* pkg/kubelet/cm will be updated to set pod-level cgroups using CPU requests and limits, and memory limits.
+* pkg/apis/core/validation/types_test.go and pkg/apis/core/validation since new fields are added in PodSpec and also new validation rules are required for the new fields.
+* pkg/kubeapiserver/admission for changes made in the admission logic for LimitRanger and ResourceQuota admission controllers.
+
+
+#### e2e tests
+
+Following scenarios need to be covered:
+
+
+
+* Cgroup settings when pod-level resources are set.
+* Validate scheduling and admission.
+* Validate the containers with no limits set are throttled on CPU when CPU usage reaches Pod level CPU limits.
+* Validate the containers with no limits set are OOMKilled when memory usage reaches Pod level memory limits.
+
+
+### Graduation Criteria
+
+
+#### Phase 1: Alpha (target 1.31)
+
+
+* Feature is disabled by default. It is an opt-in feature which can be enabled by setting the new fields in PodSpec.
```

Comment:
I don't think a feature should be enabled by stting the fields on the object the feature uses. there should be a gate and it should filter the fileds from the API request, like other KEPs

Reply:
+1 on introducing a feature gate. This provides the clear code isolation as well.

Reply:
Yep, should be Feature Gate + pod field